pkg/collector/corechecks/snmp/checkconfig/config.go

```diff
@@ -0,0 +1,482 @@
+	PrivProtocol          string            `yaml:"privProtocol"`
+	PrivKey               string            `yaml:"privKey"`
+	ContextName           string            `yaml:"context_name"`
+	Metrics               []MetricsConfig   `yaml:"metrics"`
+	MetricTags            []MetricTagConfig `yaml:"metric_tags"`
+	Profile               string            `yaml:"profile"`
+	UseGlobalMetrics      bool              `yaml:"use_global_metrics"`
+	ExtraTags             string            `yaml:"extra_tags"` // comma separated tags
+	Tags                  []string          `yaml:"tags"`       // used for device metadata
+	CollectDeviceMetadata *Boolean          `yaml:"collect_device_metadata"`
+
+	// To accept min collection interval from snmp_listener, we need to accept it as string
+	// extra_min_collection_interval can accept both string and integer value
+	MinCollectionInterval      int    `yaml:"min_collection_interval"`
+	ExtraMinCollectionInterval Number `yaml:"extra_min_collection_interval"`
+
+	// `network` config is only available in Python SNMP integration
+	// it's added here to raise warning if used with corecheck SNMP integration
+	Network string `yaml:"network_address"`
+}
+
+// CheckConfig holds config for a check instance
+type CheckConfig struct {
+	IPAddress             string
+	Port                  uint16
+	CommunityString       string
+	SnmpVersion           string
+	Timeout               int
+	Retries               int
+	User                  string
+	AuthProtocol          string
+	AuthKey               string
+	PrivProtocol          string
+	PrivKey               string
+	ContextName           string
+	OidConfig             OidConfig
+	Metrics               []MetricsConfig
+	MetricTags            []MetricTagConfig
+	OidBatchSize          int
+	BulkMaxRepetitions    uint32
+	Profiles              profileDefinitionMap
+	ProfileTags           []string
+	Profile               string
+	ProfileDef            *profileDefinition
+	ExtraTags             []string
+	InstanceTags          []string
+	CollectDeviceMetadata bool
+	DeviceID              string
+	DeviceIDTags          []string
+	Subnet                string
+	AutodetectProfile     bool
+	MinCollectionInterval time.Duration
+}
+
+// RefreshWithProfile refreshes config based on profile
+func (c *CheckConfig) RefreshWithProfile(profile string) error {
+	if _, ok := c.Profiles[profile]; !ok {
+		return fmt.Errorf("unknown profile `%s`", profile)
+	}
+	log.Debugf("Refreshing with profile `%s`", profile)
+	tags := []string{"snmp_profile:" + profile}
+	definition := c.Profiles[profile]
+	c.ProfileDef = &definition
+	c.Profile = profile
+
+	c.Metrics = append(c.Metrics, definition.Metrics...)
+	c.MetricTags = append(c.MetricTags, definition.MetricTags...)
+	c.OidConfig.addScalarOids(parseScalarOids(definition.Metrics, definition.MetricTags))
+	c.OidConfig.addColumnOids(parseColumnOids(definition.Metrics))
+
+	if definition.Device.Vendor != "" {
```

Comment:
If Device is not defined it should still work since `definition.Device` is not a pointer but struct, it's always defined, but might contain zero values for all its fields.

```
type deviceMeta struct {
	Vendor string `yaml:"vendor"`
}

type profileDefinition struct {
	Metrics      []MetricsConfig   `yaml:"metrics"`
	MetricTags   []MetricTagConfig `yaml:"metric_tags"`
	Extends      []string          `yaml:"extends"`
	Device       deviceMeta        `yaml:"device"`
	SysObjectIds StringArray       `yaml:"sysobjectid"`
}
```